docs/DeveloperGuide.adoc

```diff
@@ -239,7 +239,312 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+If you want to utilize `SearchHistoryManager` for your own use case, you can initialize a new SearchHistoryManager object
+by calling its default constructor.
+
+==== Current Implementation
+
+The main implementation behind SearchHistoryManager is a Stack Data Structure and the following 3 methods of `SearchHistoryManager` are exposed for your usage +
+
+* `executeNewSearch(Predicate<T> predicate)` +
+updates system search logic to the next state and returns a `Predicate` object storing the system search logic after the update.
+* `revertLastSearch()` +
+reverts system search logic to the previous state and returns a `Predicate` object storing
+                       the system search logic after revert.
+* `clearSearchHistory()` +
+clears all system search logic from in-app memory. +
+
+Given below are illustrations to help you understand how each method works internally
+
+ In the diagrams, UP is the short-form for User Predicate and SP is the short-form for System Predicate.
+ . User Predicate stores the logic specified by the user. This will not be used directly to filter the
+   list.
+ . System Predicate stores the search logic for the system which will be used to filter the list.
+
+NOTE: User Predicate and System Predicate are not actual Classes, they are simply there to help simplify the explanation.
+In the actual implementation, there is no way to differentiate one Predicate from the other
+
+'''
+
+`executeNewSearch(Predicate<T> predicate)` +
+
+Upon calling this method, there will be two different situations +
+
+* Situation 1: `searchHistoryStack` is empty +
+Upon receiving a new User Predicate, SearchHistoryManager will simply push the new Predicate into `searchHistoryStack`
+as a System Predicate which will be used to filter the list.
+
+image::executeNewSearchEmptyStack.png[width="800"]
+
+* Situation 2: `searchHistoryStack` is not empty +
+Before pushing the new `Predicate` into the stack, `SearchHistoryManager` will first retrieve the System Predicate object at the top of the stack.
+After retrieving it, it will call the `and()` method with the User Predicate, creating a new System Predicate which will then be pushed into the top
+of the stack.
+
+image::executeNewSearchNonEmptyStack.png[width="800"]
+'''
+`revertLastSearch()` +
+
+This method will pop the System Predicate at the top of the stack. +
+In the event that the stack is already empty, this method will throw `EmptyStackException`.
+
+image::undoSearchHistoryStack.png[width="800"]
+'''
+`clearSearchHistory()` +
+
+This method will simply empty the stack.
+
+image::clearSearchHistoryStack.png[width="800"]
+
+'''
+The following sequence diagrams shows you how `find` and `undosearch` commands utilize `SearchHistoryManager` to perform Search Pruning.
+
+ - `find` command +
+
+image::SearchPruningSequenceDiagram.png[width="800"]
+
+ - `undosearch` command +
+
+image::undoSearchSequenceDiagram.png[width="800"]
```

Comment:
Good job for drawing the sequence diagram on the following:

- drew proper lifelines for each object
- specified invocation method
- used dashed arrows for return calls.
- clear separation of Logic classes and Model classes